pallets/amm/src/lib.rs

```diff
@@ -42,33 +40,42 @@ use frame_support::{
     transactional, Blake2_128Concat, PalletId, Twox64Concat,
 };
 use frame_system::{ensure_signed, pallet_prelude::OriginFor, RawOrigin};
-use pool_structs::PoolLiquidityAmount;
-use primitives::{Balance, CurrencyId, Rate};
+pub use pallet::*;
+use primitives::{Balance, Rate};
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_runtime::traits::UniqueSaturatedInto;
+use sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedSub, Saturating};
+use sp_runtime::FixedU128;
+use sp_runtime::SaturatedConversion;
 use sp_runtime::{
-    traits::{AccountIdConversion, IntegerSquareRoot, StaticLookup},
-    ArithmeticError, DispatchError, Perbill,
+    traits::{
+        AccountIdConversion, AtLeast32BitUnsigned, IntegerSquareRoot, One, StaticLookup, Zero,
+    },
+    ArithmeticError, DispatchError, FixedPointOperand, Perbill,
 };
 pub use weights::WeightInfo;
 
+pub type AssetIdOf<T, I = ()> =
+    <<T as Config<I>>::Assets as Inspect<<T as frame_system::Config>::AccountId>>::AssetId;
+pub type BalanceOf<T, I = ()> =
+    <<T as Config<I>>::Assets as Inspect<<T as frame_system::Config>::AccountId>>::Balance;
+
 #[frame_support::pallet]
 pub mod pallet {
     use super::*;
-    use frame_support::traits::tokens::fungibles;
     use frame_system::ensure_root;
-    use primitives::CurrencyId;
+    use primitives::AssetId;
 
     #[pallet::config]
     pub trait Config<I: 'static = ()>:
-        frame_system::Config + pallet_assets::Config<AssetId = CurrencyId, Balance = Balance>
+        frame_system::Config + pallet_assets::Config<AssetId = AssetId, Balance = Balance>
```

Comment:
I don't think we need `AssetId` here. Did we try with `AssetIdOf` and `BalanceOf`?